### PR TITLE
Add translation API interface and iTranslate API

### PR DESCRIPTION
**Pull request by danielzhang130**
This is the first PR in the process of adding translation functionality to projects and comments on Catroweb. For more info, join the discussion on *#uwaterloo-project* slack channel.

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Tests - additional information
Unit tests included for `ItranslateApi`

### CODE REVIEW

File: config/packages/eight_points_guzzle.yaml [hunk 1]
```diff
@@ -3,22 +3,21 @@
 # https://github.com/8p/EightPointsGuzzleBundle/blob/master/src/Resources/doc/configuration-reference.md
 eight_points_guzzle:
     clients:
-        my_client:
```
Comment (danielzhang130) [inline config/packages/eight_points_guzzle.yaml:6]: I believe the old `my_client` was a placeholder

File: src/Translation/ItranslateApi.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    'pt-PT',
+  ];
+
+  private Client $client;
+  private string $api_key;
+
+  public function __construct(Client $client)
+  {
+    $this->client = $client;
+    $this->api_key = $_ENV['ITRANSLATE_API_KEY'];
```
Comment (dmetzner) [inline src/Translation/ItranslateApi.php:23]: Could also be injected with the service config

File: src/Translation/ItranslateApi.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        ]
+      );
+    } catch (GuzzleException $e) {
+      return null;
+    }
+
+    $statusCode = $response->getStatusCode();
+
+    if (200 != $statusCode) {
+      return null;
```
Comment (dmetzner) [inline src/Translation/ItranslateApi.php:55]: Logging such issues would help, to detect & fix issues in the future quickly.

**Review by dmetzner — APPROVED** [review]
Looking forward to the next step of your implementation! :)

**Pull request merged**